CodeEdit/Commands/CommandPaletteState.swift

```diff
@@ -0,0 +1,33 @@
+//
+//  CommandPaletteState.swift
+//  CodeEdit
+//
+//  Created by Alex on 25.05.2022.
+//
+
+import Foundation
+import Keybindings
+
+public final class CommandPaletteState: ObservableObject {
+    @Published var commandQuery: String = ""
+    @Published var selected: Command?
+    @Published var isShowingCommandsList: Bool = true
+    @Published var filteredCommands: [Command] = []
+
+    func reset() {
+        commandQuery = ""
+        selected = nil
+        filteredCommands = []
+    }
+
+    func fetchMatchingCommands(val: String) {
+        if val == "" {
+            self.filteredCommands = []
+            return
+        }
+        self.filteredCommands = CommandManager.shared.commands.filter { $0.title.localizedCaseInsensitiveContains(val) }
+        if self.filteredCommands.capacity > 0 {
```

Comment:
It's more appropriate to use `.count` to check on elements count in an array since the allocated capacity can actually be higher than the number of elements. 

And since `self.filteredCommands.first` returns an `optional` you can totally discard the `if` statement. When the array is empty this will be `nil`.